src/diffusers/models/unet_motion_model.py

```diff
@@ -0,0 +1,939 @@
+            for param in motion_modules.parameters():
+                param.requires_grad = True
+
+        motion_modules = self.mid_block.motion_modules
+        for param in motion_modules.parameters():
+            param.requires_grad = True
+
+        return
+
+    def load_motion_modules(self, motion_adapter: Optional[MotionAdapter]):
+        for i, down_block in enumerate(motion_adapter.down_blocks):
+            self.down_blocks[i].motion_modules.load_state_dict(down_block.motion_modules.state_dict())
+        for i, up_block in enumerate(motion_adapter.up_blocks):
+            self.up_blocks[i].motion_modules.load_state_dict(up_block.motion_modules.state_dict())
+
+        self.mid_block.motion_modules.load_state_dict(motion_adapter.mid_block.motion_modules.state_dict())
+
+    def save_motion_modules(
+        self,
+        save_directory: str,
+        is_main_process: bool = True,
+        safe_serialization: bool = True,
+        variant: Optional[str] = None,
+        push_to_hub: bool = False,
+        **kwargs,
+    ):
+        state_dict = self.state_dict()
+
+        # Extract all motion modules
+        motion_state_dict = {}
+        for k, v in state_dict.items():
+            if "motion_modules" in k:
+                motion_state_dict[k] = v
+
+        adapter = MotionAdapter(
+            block_out_channels=self.config["block_out_channels"],
+            motion_norm_num_groups=self.config["motion_norm_num_groups"],
+            motion_cross_attention_dim=self.config["motion_cross_attention_dim"],
+            motion_num_attention_heads=self.config["motion_num_attention_heads"],
+            motion_attention_bias=self.config["motion_attention_bias"],
+            motion_activation_fn=self.config["motion_activation_fn"],
+            motion_max_seq_length=self.config["motion_max_seq_length"],
+            motion_layers_per_block=self.config["layers_per_block"],
+        )
+        adapter.load_state_dict(motion_state_dict)
+        adapter.save_pretrained(
+            save_directory=save_directory,
+            is_main_process=is_main_process,
+            safe_serialization=safe_serialization,
+            variant=variant,
+            push_to_hub=push_to_hub,
+            **kwargs,
+        )
+
+    @property
+    # Copied from diffusers.models.unet_2d_condition.UNet2DConditionModel.attn_processors
+    def attn_processors(self) -> Dict[str, AttentionProcessor]:
+        r"""
+        Returns:
+            `dict` of attention processors: A dictionary containing all attention processors used in the model with
+            indexed by its weight name.
+        """
+        # set recursively
+        processors = {}
+
+        def fn_recursive_add_processors(name: str, module: torch.nn.Module, processors: Dict[str, AttentionProcessor]):
+            if hasattr(module, "get_processor"):
+                processors[f"{name}.processor"] = module.get_processor(return_deprecated_lora=True)
```

Comment:
```suggestion
                processors[f"{name}.processor"] = module.get_processor()
```
Let's make sure to not copy deprecated functionality